src/smtsolvers/ParallelScatterSplitter.h

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2022, Antti Hyvarinen <antti.hyvarinen@gmail.com>
+ * Copyright (c) 2022, Seyedmasoud Asadzadeh <seyedmasoud.asadzadeh@usi.ch>
+ *
+ * SPDX-License-Identifier: MIT
+ */
+#ifndef OPENSMT_PARALLELSCATTERSPLITTER_H
+#define OPENSMT_PARALLELSCATTERSPLITTER_H
+
+#include "ScatterSplitter.h"
+
+#include <PTPLib/net/Channel.hpp>
+#include <PTPLib/common/Exception.hpp>
+#include <PTPLib/common/Printer.hpp>
+
+class ParallelScatterSplitter : public ScatterSplitter  {
+public:
+    ParallelScatterSplitter(SMTConfig & c, THandler & t);
+    void set_channel(PTPLib::net::Channel & ch) { channel = &ch; }   //SMTS Client owns the channel and should directly set the channel in ParallelScatterSpitter
+    PTPLib::net::Channel & getChannel() const { return *channel;};
+    void set_syncedStream(PTPLib::common::synced_stream & ss) { syncedStream = &ss; }  //SMTS Client owns the SyncedStream and should directly set the SyncedStream in ParallelScatterSpitter
+    vec<opensmt::pair<int,int>> const &  get_solver_branch()  const  { return solverBranch; }
+    void set_solver_branch(std::string solver_branch);
```

Comment:
There are some strange semicolons at the end of the declarations in some places here.

Reply:
And below as well.